### DISCUSSION TITLE
Atmos workflow for ARC EKS automation cluster

### DISCUSSION THREAD

**Discussion by rauthur** [cloudposse/community#19] [Essential Support] [answered]
README steps document automation cluster deployment as:

```
atmos workflow cluster/deploy -s core-usw2-auto -f eks
atmos workflow auto/deploy -f eks
```

The first should be `deploy/cluster` I think. The second doesn't seem to exist in the workflow (either as `auto/deploy` or `deploy/auto`). What are the workflow steps for `auto/deploy` (or for `deploy/auto` if the naming is reversed)?

**Comment by rauthur** [comment]
I think it possibly just needs the `eks/actions-runner-controller` component but figured it's worth posting.

**Reply by GabisCampana** [reply]
@dudymas @Nuru

**Comment by milldr** [comment, marked as answer]
Ah thanks for point that out; yes it's a mistake. There are a few ways to deploy it, but the docs should say this 1 command to do it all:

```bash
atmos workflow deploy/github-runners -f github
```

This command will do everything you need:
1. Provision service IAM roles for core-auto
2. Deploy the core-auto EKS cluster
3. Deploy all standard EKS resources
4. Then finally deploy `eks/actions-runner-controller`

Ultimately it's the same as any other `plat` EKS cluster, but for self-hosted runners we deploy `eks/actions-runner-controller` after everything else is deployed

**Reply by milldr** [reply]
That command would be the same thing as this, if you want to break it down:

```bash
atmos terraform deploy iam-service-linked-roles -s core-use1-auto
atmos workflow deploy/cluster -s core-use1-auto -f eks
atmos workflow deploy/resources -f eks
atmos terraform deploy eks/actions-runner-controller -s core-use1-auto
```

**Reply by rauthur** [reply]
Thanks, I think there is a minor difference in that `deploy/github-runners` includes `workflow deploy/pl-github-runners -f github` which I don't think is necessary with EKS runners.

**Reply by milldr** [reply]
Yes that's right. "pl-github-runners" refers to Philips Labs runners, which is an alternative to EKS ARC runners when customers are not deploying EKS
